packages/@lwc/engine-core/src/framework/template.ts

```diff
@@ -224,6 +229,22 @@ export function evaluateTemplate(vm: VM, html: Template): VNodes {
     return vnodes;
 }
 
+function validateTemplateVersion(template: Template, vm: VM) {
+    // Validate that the template was compiled with the same version of the LWC compiler used for the runtime engine.
+    // Note that 1 is a special version used to identify the ACT Compiler, and we continue to maintain support for
+    // templates with this version.
+    assert.isTrue(
+        template === defaultEmptyTemplate ||
+            template.version === 1 ||
+            template.version === COMPILER_VERSION_NUMBER,
+        `Mismatch between compiled template version and runtime engine version. Current engine is v${COMPILER_VERSION_NUMBER}, but ${getComponentTag(
+            vm
+        )} was compiled with ${
+            isUndefined(template.version) ? 'an unknown version' : `v${template.version}`
+        }.`
+    );
+}
```

Comment:
A pragmatic option: get rid of this assertion entirely, and add it back in later when we need to bump the version from 2 to 3. It's probably too late for us to warn anybody about templates older than the current version.